masonry/src/passes/update.rs

```diff
@@ -145,3 +145,26 @@ pub(crate) fn run_update_pointer_pass(root: &mut RenderRoot, root_state: &mut Wi
     // Pass root widget state to synthetic state create at beginning of pass
     root_state.merge_up(root.widget_arena.get_state_mut(root.root.id()).item);
 }
+
+// ----------------
+
+pub(crate) fn run_update_scroll_pass(root: &mut RenderRoot) {
+    let _span = info_span!("update_scroll").entered();
+
+    let scroll_request_targets = std::mem::take(&mut root.state.scroll_request_targets);
+    for (target, rect) in scroll_request_targets {
+        let mut target_rect = rect;
+
+        run_targeted_update_pass(root, Some(target), |widget, ctx| {
+            let event = LifeCycle::RequestPanToChild(rect);
+            widget.lifecycle(ctx, &event);
+
```

Comment:
How will smooth scroll would be handled?

The best logic I can see if to "get where the widget *will be*", and all the parents scroll to that.

Reply:
I don't mean on a touchscreen, I mean scrolling to a position over a certain amount of time.

Right, yeah. This logic would have to change at that point, but it's also deliberately unaddressed at the moment.